splash/browser_tab.py

```diff
@@ -827,11 +839,21 @@ def get(self, url, callback, headers=None, follow_redirects=True):
 
     def _send_request(self, url, callback, method='GET', body=None,
                       headers=None):
-        # XXX: The caller must ensure self._delete_reply is called in a callback.
-        if method != 'GET':
-            raise NotImplementedError()
-        request = self.request_obj(url, headers=headers)
-        reply = self.network_manager.get(request)
+        # XXX: The caller must ensure self._delete_reply is called in a callback
+        request = self.request_obj(url, headers=headers, body=body)
+
+        if method.upper() not in OPERATION_QT_CONSTANTS:
+            raise ValueError("Invalid HTTP method: '{}'".format(method))
+
+        operation = getattr(self.network_manager, method.lower())
```

Comment:
I'm not sure about this method, docs say:

> This method provides means to send verbs other than the common ones provided via get() or post() etc., for instance sending an HTTP OPTIONS command.

so I guess this would be for some uncommon HTTP like OPTIONS.

We could use [createRequest method of network manager](http://doc.qt.io/qt-5/qnetworkaccessmanager.html#createRequest) but then again I wonder if we really should add support for all methods in this one PR.

I initially thought that it could be useful and not difficult to support all verbs, but now I see there are many edges cases here. Aside from one you mentioned seems like DELETE methods [may have a body](http://stackoverflow.com/questions/299628/is-an-entity-body-allowed-for-an-http-delete-request).

What do you think about limiting this PR to adding POST only? I would refactor this PR and limit allowed HTTP verbs to POST and GET. We could create new PR for handling all other HTTP methods, this way we could also add separate unit tests for all other methods and we could look into all edge cases related to other HTTP methods.